src/zimscraperlib/download.py

```diff
@@ -7,17 +7,33 @@
 
 import requests
 
+from . import logger
+
 WGET_BINARY = os.getenv("WGET_BINARY", "/usr/bin/wget")
 
 
-def save_file(url, fpath):
-    """ download a binary file from its URL """
-    req = requests.get(url)
-    req.raise_for_status()
-    if not fpath.parent.exists():
-        fpath.parent.mkdir(exist_ok=True)
-    with open(fpath, "wb") as fp:
-        fp.write(req.content)
+def save_file(url, fpath, timeout=30, retries=0):
+    """ download a file from its URL, and return headers
+
+        Only recommended to be used with small files/HTMLs """
+
+    max_attempts = retries + 1
+    for attempt in range(max_attempts):
+        try:
+            resp = requests.get(url, timeout=timeout)
+            resp.raise_for_status()
+            if not fpath.parent.exists():
+                fpath.parent.mkdir(exist_ok=True)
+            with open(fpath, "wb") as fp:
+                fp.write(resp.content)
+            return resp.headers
+        except requests.exceptions.RequestException as exc:
+            if retries == 0:
```

Comment:
I don't like that we have two behaviors for this function based on the number of retries requested.
Also, it you use retries, then you can't catch the proper exception

Reply:
Okay. It re-throws the correct exception now

Reply:
Okay, now re-throws the original exception